tpm2/tpm2.go

```diff
@@ -906,24 +906,42 @@ func encodeQuote(signingHandle tpmutil.Handle, parentPassword, ownerPassword str
 	return concat(ha, auth, params, pcrs)
 }
 
-func decodeQuote(in []byte) ([]byte, *Signature, error) {
+func decodeQuote(in []byte) ([]byte, []byte, error) {
 	buf := bytes.NewBuffer(in)
 	var paramSize uint32
 	var attest tpmutil.U16Bytes
 	if err := tpmutil.UnpackBuf(buf, &paramSize, &attest); err != nil {
 		return nil, nil, err
 	}
-	sig, err := DecodeSignature(buf)
-	return attest, sig, err
+	// the unread part of the buf is the raw signature
```

Comment:
I don't think this line is correct. I think we would have to use `paramSize` to figure out where the remaining buffer ends.

See Part 1, Section 18.1 pf the spec

Reply:
Calculated the signature buf size based on paramSize and attestation buf size.